db/compaction/compaction.cc

```diff
@@ -763,7 +764,16 @@ int Compaction::EvaluatePenultimateLevel(
 
   int penultimate_level = output_level - 1;
   assert(penultimate_level < immutable_options.num_levels);
-  if (penultimate_level <= 0 || penultimate_level < start_level) {
+  if (penultimate_level <= 0) {
+    return kInvalidLevel;
+  }
+
+  // If the penultimate level is not within input level -> output level range
+  // check if the penultimate output level is empty, if it's empty, it could
+  // also be locked for the penultimate output.
+  if (penultimate_level < start_level &&
```

Comment:
I feels more straight forward to have the condition to be `start_level ==immutable_options.num_levels - 1` or `start_level == output_level`. I imagine that it means the same.

Reply:
Updated.